### ISSUE TITLE
Microsoft.Web.WebView2 versions after 1.0.1245.22 break EF Core Scaffold-DbContext

### ISSUE THREAD

**Issue by NerdArray** [MicrosoftEdge/WebView2Feedback#2686]
After upgrading Microsoft.Web.WEbView2 past version 1.0.1245.22 (using NuGet), I am no longer able to run Scaffold-DbContext from within my project.  
```
PM> Scaffold-DbContext -Connection "Data Source=DEVBOX\DEVSQL;Initial Catalog=CCPC_MAIN;Trusted_Connection=True" -Provider Microsoft.EntityFrameworkCore.SqlServer -OutputDir "Models\EF" -ContextDir "Models\EF" -Context ApplicationDbContext -DataAnnotations -UseDatabaseNames -Force -NoOnConfiguring -Project CCPC.RecordDB -Namespace CCPC.RecordDB.Models.EF -ContextNamespace CCPC.RecordDB.Models.EF
Build started...
Build succeeded.
Exception calling ".ctor" with "1" argument(s): "The parameter 'frameworkName' cannot be an empty string.
Parameter name: frameworkName"
```

Reverting back to 1.0.1245.22 resolves the issue.  I can reproduce the issue with v1.0.1264.42 and v1.0.1293.44.

From my csproj file...
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
		<TargetFramework>net6.0-windows10.0.22000.0</TargetFramework>
    <UseWPF>true</UseWPF>
    <SupportedOSPlatformVersion>10.0.17763.0</SupportedOSPlatformVersion>
    <PlatformTarget>x64</PlatformTarget>
    <ApplicationIcon>CCPC.ico</ApplicationIcon>
    <StartupObject>CCPC.RecordDB.App</StartupObject>
  </PropertyGroup>
...
<ItemGroup>
    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.8">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="6.0.8">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
    <PackageReference Include="Microsoft.Graph" Version="4.36.0" />
    <PackageReference Include="Microsoft.Identity.Client" Version="4.46.0" />
    <PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Version="2.23.0" />
    <PackageReference Include="Microsoft.PowerBI.Api" Version="4.9.0" />
    <PackageReference Include="Microsoft.Web.WebView2" Version="1.0.1245.22" />
  </ItemGroup>
...
</Project>
```

[AB#41077851](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/41077851)

**Comment by LiangTheDev** [comment]
Could you try to add `<WebView2UseWinRT>false</WebView2UseWinRT>` into the `<PropertyGroup>` and see whether it changes anything? And which version of the VS are you using?

**Comment by NerdArray** [comment]
I added `<WebView2UseWinRT>false</WebView2UseWinRT>` to the `<PropertyGroup>` as requested.  Scaffold-DbContext still works as expected when using version 1.0.1245.22:
`<PackageReference Include="Microsoft.Web.WebView2" Version="1.0.1245.22" />`

However, the issue still returns after updating the version to either 1.0.1264.42 or 1.0.1293.44.  

I am currently using VS 17.3.1, but the issue also occurred in VS v17.3.0 and v17.2.6.

**Comment by Syul968** [comment]
Hi @NerdArray,

The WebView2 SDK 1.0.1264.42 introduces a properties page and new targets. We have seen some seemingly unrelated issues from it. Would you try the following:

In your project file, right after the WebView2 SDK targets import, add:
```
<ItemGroup>
  <PropertyPageSchema Remove="$(NugetRoot)build\WebView2Rules.Project.xml" />
</ItemGroup>
```

If this doesn't fix the issue, you could try removing the new targets; from `<SDK>\build\Common.targets`, remove line 76:
```
<Import Project="$(NugetRoot)\build\wv2winrt.targets"/>
```

Please let me know if either of these helps fix the issue.

**Comment by NerdArray** [comment]
I think my project file is missing the SDK target.  The second option you mentioned didn't work.  I removed line 76 from `C:\Users\Chris\.nuget\packages\microsoft.web.webview2\1.0.1293.44\build\Common.targets`, but it didn't help.

Here is my full project file....

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
		<TargetFramework>net6.0-windows10.0.22000.0</TargetFramework>
    <UseWPF>true</UseWPF>
    <SupportedOSPlatformVersion>10.0.17763.0</SupportedOSPlatformVersion>
    <PlatformTarget>x64</PlatformTarget>
    <ApplicationIcon>CCPC.ico</ApplicationIcon>
    <StartupObject>CCPC.RecordDB.App</StartupObject>
    <WebView2UseWinRT>false</WebView2UseWinRT>
  </PropertyGroup>
  
  <ItemGroup>
    <Compile Remove="Views\Controls\**" />
    <EmbeddedResource Remove="Views\Controls\**" />
    <None Remove="Views\Controls\**" />
    <Page Remove="Views\Controls\**" />
  </ItemGroup>

  <ItemGroup>
    <None Remove="appsettings.json" />
    <None Remove="CCPC_Splash.png" />
    <None Remove="Templates\ConsentFormTemplate.dotx" />
    <None Remove="Views\Reporting\ReportLoader.html" />
    <None Remove="ReportWeb\Scripts\ReportLoader.js" />
  </ItemGroup>

  <ItemGroup>
    <COMReference Include="Microsoft.Office.Interop.Word">
      <WrapperTool>tlbimp</WrapperTool>
      <VersionMinor>7</VersionMinor>
      <VersionMajor>8</VersionMajor>
      <Guid>00020905-0000-0000-c000-000000000046</Guid>
      <Lcid>0</Lcid>
      <Isolated>false</Isolated>
      <EmbedInteropTypes>true</EmbedInteropTypes>
    </COMReference>
    <COMReference Include="Microsoft.Office.Interop.Excel">
      <WrapperTool>tlbimp</WrapperTool>
      <VersionMinor>9</VersionMinor>
      <VersionMajor>1</VersionMajor>
      <Guid>00020813-0000-0000-c000-000000000046</Guid>
      <Lcid>0</Lcid>
      <Isolated>false</Isolated>
      <EmbedInteropTypes>true</EmbedInteropTypes>
    </COMReference>
  </ItemGroup>

  <ItemGroup>
    <Content Include="appsettings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
    <Content Include="CCPC.ico" />
    <Content Include="Templates\ConsentFormTemplate.dotx">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="ReportWeb\ReportLoader.html">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
    <Content Include="ReportWeb\Scripts\ReportLoader.js">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
  </ItemGroup>

  <ItemGroup>
    <Folder Include="Models\EF\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="libphonenumber-csharp" Version="8.12.53" />
    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="6.0.8">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.8" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="6.0.8">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="6.0.0" />
    <PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
    <PackageReference Include="Microsoft.Graph" Version="4.36.0" />
    <PackageReference Include="Microsoft.Identity.Client" Version="4.46.1" />
    <PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Version="2.23.0" />
    <PackageReference Include="Microsoft.PowerBI.Api" Version="4.9.0" />
    <PackageReference Include="Microsoft.Web.WebView2" Version="1.0.1293.44" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\CCPC.Styling.Dark\CCPC.Styling.Dark.csproj" />
    <ProjectReference Include="..\CCPC.Styling.Default\CCPC.Styling.Default.csproj" />
  </ItemGroup>

  <ItemGroup>
    <Reference Include="BingMapsRESTToolkit">
      <HintPath>..\..\..\BingMapsRESTToolkit\Source\bin\Release\BingMapsRESTToolkit.dll</HintPath>
    </Reference>
  </ItemGroup>

  <ItemGroup>
    <SplashScreen Include="CCPC_Splash.png" />
  </ItemGroup>

  <ItemGroup>
    <Compile Update="Properties\UserSettings.Designer.cs">
      <DesignTimeSharedInput>True</DesignTimeSharedInput>
      <AutoGen>True</AutoGen>
      <DependentUpon>UserSettings.settings</DependentUpon>
    </Compile>
  </ItemGroup>

  <ItemGroup>
    <None Update="Properties\UserSettings.settings">
      <Generator>SettingsSingleFileGenerator</Generator>
      <LastGenOutput>UserSettings.Designer.cs</LastGenOutput>
    </None>
  </ItemGroup>

  <ProjectExtensions><VisualStudio><UserProperties /></VisualStudio></ProjectExtensions>

</Project>
```

**Comment by Syul968** [comment]
The intention of the first change is to prevent the properties page to load for the project. Since this is an SDK-style project using `PackageReference` instead of `packages.config` + targets imports, you could try to change that `Common.targets` file to remove the properties page include in the first place, which is done in lines 17-19:

```
  <ItemGroup>
    <PropertyPageSchema Include="$(MSBuildThisFileDirectory)\WebView2Rules.Project.xml" Condition="'$(VisualStudioVersion)' &gt;= '17.0' or '$(WebView2ProjectKind)' == 'native'"/>
  </ItemGroup>
```

You could also check that the changes are reflected in the copied targets file (if any) to the `obj` directory for the project. This is just to check whether the properties page import is the culprit. Alternatively, if you are able to use VS2019, you could try that instead as the import for managed projects is skipped in versions below VS2022.

Also, if you have a sample project where the issue is reproduced, feel free to share that here.

**Comment by NerdArray** [comment]
Thank you for your help.  I removed lines 17-19 of the `Common.targets` file and deleted the bin/obj directories in my project folder.  Afterwards, `Scaffold-DbContext` worked as expected.

I also created a test project and test database where I was able to reproduce the issue after uncommenting lines 17-19 in the `Common.targets` file, and running `Scaffold-DbContext "Data Source=DEVBOX\DEVSQL;Initial Catalog=Test;Trusted_Connection=True;TrustServerCertificate=True" Microsoft.EntityFrameworkCore.SqlServer -Context TestDbContext -DataAnnotations -UseDatabaseNames -Force -NoOnConfiguring` from the Package Manager console. 

[WebViewTest.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/9383600/WebViewTest.zip)
